src/components/layout/MainMenu/LeftMenu.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect, useState } from 'react';
+import { Link, useRouteMatch } from 'react-router-dom';
+import { Menu, Grid } from 'antd';
+
+const { useBreakpoint } = Grid;
+
+const LeftMenu = () => {
+  const { lg } = useBreakpoint();
+  const [selectedKeys, setSelectedKeys] = useState();
+
+  const isDac = useRouteMatch('/dac/:slug') || useRouteMatch('/dacs');
+  const isCamp = useRouteMatch('/campaign/:slug') || useRouteMatch('/campaigns');
+  const isMilestone = useRouteMatch('/milestone/:slug') || useRouteMatch('/milestones');
+
+  let selectedKey;
+  if (isDac) selectedKey = 'Communities';
+  else if (isCamp) selectedKey = 'Campaigns';
+  else if (isMilestone) selectedKey = 'Milestones';
+  else selectedKey = '';
+
+  useEffect(() => {
+    setSelectedKeys(selectedKey);
+  }, [selectedKey]);
```

Comment:
Make this useEffect dependent to path. Maybe location parameter passed by `withRouter`
```suggestion
  }, [props.location]);
```

> A location object is never mutated so you can use it in the lifecycle hooks to determine when navigation happens, this is really useful for data fetching and animation.

If above works, all computation of other variables should be happen in useEffect, I mean selectedKey, isDac and etc.

>

Reply:
I tried different solutions like useRouteMatch, useLocation and useParams but none of them can be defined inside useEffect because hooks must be defined inside body of functional components.